File: 1913-maximum-product-difference-between-two-pairs/1913-maximum-product-difference-between-two-pairs.cpp
class Solution {
public:
    int maxProductDifference(vector<int>& nums) {
        int ma1 = nums[0], ma2 = nums[1], mi1 = nums[2], mi2 = nums[3];
        vector<int> four = {ma1, ma2, mi1, mi2};
        sort(four.begin(), four.end());
        mi1 = four[0];
        mi2 = four[1];
        ma1 = four[3];
        ma2 = four[2];
    
        for(int i=4; i<nums.size(); i++){
            int ele = nums[i];
            
            // should max be updated
            if(ele > ma2 || ele > ma1){
                if(ele > ma1){
                    ma2 = ma1;
                    ma1 = ele;
                }else
                    ma2 = ele;
            }
            
            //should min be updated
            if(ele < mi2 || ele < mi1){
                if(ele < mi1){
                    mi2 = mi1;
                    mi1 = ele;
                }else
                    mi2 = ele;
            }
            
        }
        
        int first = mi1 * mi2;
        int sec = ma1 * ma2;
        return -( first - sec);
    }
};

File: 1913-maximum-product-difference-between-two-pairs/1913-maximum-product-difference-between-two-pairs.py
class Solution:
    def maxProductDifference(self, nums: List[int]) -> int:
        four = [0] * 4
        for i in range(0,4):
            four[i] = nums[i]
        four = sorted(four)
        mi1, mi2, ma2, ma1 = four[0], four[1], four[2], four[3]
        
        for i in range(4, len(nums)):
            ele = nums[i]
            if(ele >= ma2 or ele >= ma1):
                if(ele > ma1):
                    ma2 = ma1
                    ma1 = ele
                else:
                    ma2 = ele
                    
            elif (ele <= mi2 or ele <= mi1):
                if(ele < mi1):
                    mi2 = mi1
                    mi1 = ele
                else:
                    mi2 = ele
                    
        return (ma1*ma2) - (mi1*mi2)